packages/block-editor/src/components/block-popover/index.js

```diff
@@ -63,6 +63,7 @@ export default function BlockPopover( {
 			__unstableObserveElement={ selectedElement }
 			__unstableForcePosition
 			__unstableShift
+			__unstableAvoidOverflow
```

Comment:
Yeah, I agree that it'd be great to avoid these props.

Long term, I think floating-ui's `middleware` API could serve as an inspiration. It'd be nice to have something like it that's not quite as low level. Popover positioning could be more about composition of functions or hooks rather than boolean props:
```js
function useBlockPopoverPositioning( selectedElement ) {
    return [
        usePopoverObservePosition( selectedElement ),
        usePopoverForcePosition(),
        usePopoverShift(),
    ];
}

function BlockPopover( selectedElement, // ... ) {
    // ...
    
    return (
        <Popover
            positioning={ useBlockPopoverPositioning( selectedElement ) }
            // ...
        />
    );
}
```